numpy/core/src/umath/ufunc_object.c

```diff
@@ -2089,6 +2107,124 @@ PyUFunc_GeneralizedFunction(PyUFuncObject *ufunc,
         goto fail;
     }
 
+    if (axis) {
+        /*
+         * possibly remap axes.
+         */
+        int axis_rqst;
+        int axis_list = PyList_Check(axis);
```

Comment:
I believe `axis` on `np.sum` actually only allows tuples, not lists -- it would be good to be consistent.